core/src/main/java/org/apache/accumulo/core/fate/user/UserFateStore.java

```diff
@@ -151,6 +157,95 @@ protected void create(FateId fateId, FateKey fateKey) {
         + " and fateKey " + fateKey + " after " + maxAttempts + " attempts");
   }
 
+  @Override
+  public Optional<FateTxStore<T>> tryReserve(FateId fateId) {
+    // Create a unique FateReservation for this reservation attempt
+    FateReservation reservation = FateReservation.from(lockID, UUID.randomUUID());
+
+    FateMutator.Status status = newMutator(fateId).putReservedTx(reservation).tryMutate();
+    if (status.equals(FateMutator.Status.ACCEPTED)) {
+      return Optional.of(new FateTxStoreImpl(fateId, reservation));
+    } else if (status.equals(FateMutator.Status.UNKNOWN)) {
+      // If the status is UNKNOWN, this means an error occurred after the mutation was
+      // sent to the TabletServer, and it is unknown if the mutation was written. We
+      // need to check if the mutation was written and if it was written by this
+      // attempt at reservation. If it was written by this reservation attempt,
+      // we can return the FateTxStore since it was successfully reserved in this
+      // attempt, otherwise we return empty (was written by another reservation
+      // attempt or was not written at all).
+      try (Scanner scanner = context.createScanner(tableName, Authorizations.EMPTY)) {
+        scanner.setRange(getRow(fateId));
+        scanner.fetchColumn(TxColumnFamily.RESERVATION_COLUMN.getColumnFamily(),
+            TxColumnFamily.RESERVATION_COLUMN.getColumnQualifier());
+        FateReservation persistedRes = scanner.stream()
+            .filter(entry -> FateReservation.isFateReservation(entry.getValue().toString()))
+            .map(entry -> FateReservation.from(entry.getValue().toString())).findFirst()
+            .orElse(null);
+        if (persistedRes != null && persistedRes.equals(reservation)) {
+          return Optional.of(new FateTxStoreImpl(fateId, reservation));
+        }
```

Comment:
```suggestion
        return scanner.stream().map(entry -> entry.getValue().toString())
            .filter(FateReservation::isFateReservation).map(FateReservation::from)
            .filter(persistedRes -> persistedRes.equals(reservation)).findFirst()
            .map(persistedRes -> new FateTxStoreImpl(fateId, reservation));
```

These could be combined

Reply:
I wrote what would be the equivalent stream needed, and it was pretty complicated. I personally think the way it is now is a bit more readable/easy to follow. Since it would be the same, I'm gonna leave it as is.